app/uk/gov/hmrc/agentaccesscontrol/service/MtdAuthorisationService.scala

```diff
@@ -24,16 +26,37 @@ import uk.gov.hmrc.play.http.HeaderCarrier
 import scala.concurrent.{ExecutionContext, Future}
 
 class MtdAuthorisationService(agenciesConnector: AgenciesConnector,
-                              relationshipsConnector: RelationshipsConnector) {
+                              relationshipsConnector: RelationshipsConnector,
+                              auditService: AuditService) extends LoggingAuthorisationResults {
 
-  def authoriseForSa(agentCode: AgentCode, mtdSaClientId: MtdSaClientId)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Boolean] = {
+  def authoriseForSa(agentCode: AgentCode, mtdSaClientId: MtdSaClientId)
+                    (implicit ec: ExecutionContext, hc: HeaderCarrier, request: Request[_]): Future[Boolean] = {
     agenciesConnector.fetchAgencyRecord(agentCode) flatMap  {
-      case Some(agency) => hasRelationship(agency.arn, mtdSaClientId)
-      case None => Future successful false
+      case Some(agency) => hasRelationship(agency.arn, mtdSaClientId) map { result =>
+        auditDecision(agentCode, mtdSaClientId, result, "arn" -> agency.arn)
```

Comment:
I think it would be nice to test MtdAuthorisationService's auditing in a unit test not just an integration test (AuthorisationService's auditing is already unit tested in AuthorisationServiceSpec - admittedly not in the nicest possible way in that the asserts are slipped into other tests not done in a "one assert per test" style).